pkg/imgpkg/registry/registry.go

```diff
@@ -34,17 +34,64 @@ type Opts struct {
 	Anon     bool
 
 	ResponseHeaderTimeout time.Duration
+
+	EnvironFunc func() []string
+}
+
+// Registry Interface to access the registry
+type Registry interface {
+	Get(reference regname.Reference) (*regremote.Descriptor, error)
+	Digest(reference regname.Reference) (regv1.Hash, error)
+	Index(reference regname.Reference) (regv1.ImageIndex, error)
+	Image(reference regname.Reference) (regv1.Image, error)
+	FirstImageExists(digests []string) (string, error)
+
+	MultiWrite(imageOrIndexesToUpload map[regname.Reference]regremote.Taggable, concurrency int, updatesCh chan regv1.Update) error
+	WriteImage(reference regname.Reference, image regv1.Image) error
+	WriteIndex(reference regname.Reference, index regv1.ImageIndex) error
+	WriteTag(tag regname.Tag, taggable regremote.Taggable) error
+
+	ListTags(repo regname.Repository) ([]string, error)
+
+	CloneWithSingleAuth(imageRef regname.Tag) (Registry, error)
+}
+
+// ImagesMetadata Interface for Reading Images
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . ImagesMetadata
+type ImagesMetadata interface {
```

Comment:
should this be called ImagesReader to be consistent with ImagesReaderWriter naming?

Reply:
Fixed